routes/search.js

```diff
@@ -0,0 +1,27 @@
+let bcrypt = require('bcrypt')
+let User = require('../models/User')
+let jwt = require('jsonwebtoken')
+
+module.exports = function (express) {
+  let router = express.Router()
+  
+  router.route('/') 
+    // requires user. Adds post to req.user
+    .get((req, res, next) => {
+      // check auth
+      if (!req.user) { return res.status(401).json({ error: 'Unauthorized' })}
+      console.log(`[garden] GET /api/search:`, req.query)
+      let query = {}
+      let searchTerm = req.query.displayName || req.query.username || ''
```

Comment:
Even if we don't open it up to that extent, we may want to at least add a `flag` or two. In the cases you're searching specifically for people you're `following` or are `followers`. That way we can narrow the results down to different pages.